### PR TITLE
Matomo metrics

### PR DESCRIPTION
**Pull request by danigm**
This is still work in progress. We need a real matomo server to be able to use this. And the data sent is just an example, we need to take a look to the API and see if there's better data that we can use.

https://phabricator.endlessm.com/T30942

### CODE REVIEW

**Review by manuq — CHANGES_REQUESTED** [review]
Looking good. Let's remove eosmetrics altogether.

File: meson_options.txt [hunk 1]
```diff
@@ -8,3 +8,21 @@ option('run-lint',
        type: 'boolean',
        value: true
 )
+
+option('eosmetrics',
+       description: 'whether to use eosmetrics or not',
+       type: 'boolean',
+       value: true
+)
```
[the remark this answers is not in the document]
Comment (danigm) [reply meson_options.txt]: I've added some new commits to remove the eosmetrics from the clubhouse and also to improve the data sent to the matomo server.

I've also added a new method to the clubhouse dbus API to use in other apps, like the toolbox, to be able to register metrics on matomo from other components.

There's also another PR related to this one, to use the new API instead of the eosmetrics on the toolbox:     https://github.com/endlessm/hack-toolbox-app/pull/250

File: katamari/tools/build-clubhouse [hunk 2]
```diff
@@ -41,6 +41,8 @@ DEFAULTS = {
     'clubhouse': {
         'run_lint': True,
         'run_tests': True,
+        'matomo_host': 'http://matomo.test.endlessos.org',
```
Comment (manuq) [inline katamari/tools/build-clubhouse:44]: `http://`... No secure communication?

File: katamari/com.hack_computer.Clubhouse.json.in [hunk 3]
```diff
@@ -234,6 +179,21 @@
...
+                {
+                    "type": "git",
+                    "url": "https://github.com/peter-wangxu/persist-queue.git",
+                    "tag": "v0.5.0",
+                    "commit": "f27b0ca8b52454a79de8cf4e1f7abceed6307def"
+                }
+            ],
+            "build-commands": [
+                "python3 ./setup.py install --prefix=/app"
+            ]
```
Comment (manuq) [inline katamari/com.hack_computer.Clubhouse.json.in:195]: :+1:

File: eosclubhouse/utils.py [hunk 4]
```diff
@@ -390,17 +390,32 @@ def get_flatpak_sandbox():
...
+        signature += ')'
+        return signature
+
+    if isinstance(variant, list):
+        signature = 'a' + get_variant_signature(variant[0])
+        return signature
+
+    raise TypeError('Error: value is not a basic type')
+
+
```
Comment (manuq) [inline eosclubhouse/utils.py:413]: Can you explain why are these changes needed? If they were only for eosmetrics, I'd left them out of the PR.

File: eosclubhouse/system.py [hunk 5]
```diff
@@ -1206,6 +1206,15 @@ def get_chassis(klass):
         prop = klass.proxy().get_cached_property('Chassis')
         return prop.unpack()
 
+    @classmethod
+    def get_os(klass):
+        prop = klass.proxy().get_cached_property('OperatingSystemPrettyName')
+        os_name = prop.unpack()
+        os_version = ''
+        if ' ' in os_name:
+            os_name, os_version = os_name.split(' ')
```
Comment (manuq) [inline eosclubhouse/system.py:1215]: This will fail to unpack it there are 2 or more space characters in `os_name`. Are you confident that's not the case?
```
>>> foo, bar = 'some random string'.split(' ')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
```

Anyways it would be safer to do:
```
if ' ' in os_name:
    return os_name.split(' ')[0], os_name.split(' ')[1]
```

**Review by manuq — APPROVED** [review]
Looks good! It only needs a rebase, because latter commits erase things added in previous commits. For instance USE_EOSMETRICS is added in commit "Move metrics to new metrics module" and then is removed in commit "metrics: Remove eosmetrics". We should keep the git history clean. Please feel free to merge after rebasing.

File: eosclubhouse/utils.py (hunk 4 above)
Comment (danigm) [reply eosclubhouse/utils.py:413]: Yes, you're right, this is not needed, so we can remove these changes from the PR.

File: eosclubhouse/system.py (hunk 5 above)
Comment (danigm) [reply eosclubhouse/system.py:1215]: I've only checked the name on ubuntu, fedora and endless and there there's no problem, but as you said, this is a bit fragile. I'll try to improve this to make it more robust because I think that this name could be anything and there are a lot of linux distributions.

File: katamari/tools/build-clubhouse (hunk 2 above)
Comment (danigm) [reply katamari/tools/build-clubhouse:44]: it's a test server, it'll be a secure one once we've the production one.
